### PR TITLE
Added Amazon Redshift dialect and timestamp/date add functions

### PR DESCRIPTION
**Pull request by mikeengland**

### CODE REVIEW

File: pypika/terms.py [hunk 1]
```diff
@@ -847,7 +849,7 @@ def fields(self):
         return []
 
     def get_sql(self, **kwargs):
-        dialect = kwargs.get('dialect')
+        dialect = self.dialect or kwargs.get('dialect')
```
Comment (twheys) [inline pypika/terms.py]: does it make sense to have two sources for dialect? Where do each of them come from? Is it not redundant?

File: pypika/terms.py [hunk 2]
```diff
@@ -880,6 +882,10 @@ def get_sql(self, **kwargs):
             # MySQL requires no single quotes around the expr and unit
             Dialects.MYSQL: 'INTERVAL {expr} {unit}',
 
+            # PostgreSQL and Redshift require quotes around the expr and unit e.g. INTERVAL '1 week'
+            Dialects.POSTGRESQL: 'INTERVAL \'{expr} {unit}\'',
+            Dialects.REDSHIFT: 'INTERVAL \'{expr} {unit}\'',
```
Comment (twheys) [inline pypika/terms.py]: Could you please add some test coverage for this?

File: pypika/terms.py (hunk 1 above)
Comment (mikeengland) [reply pypika/terms.py]: The dialect gets passed to the get_sql() function when the interval is used inside another function. However, just using Interval() by itself doesn't receive the dialect, meaning the default (and incorrect) interval SQL was being returned for the MySQL platform when I was testing.

Comment (twheys) [reply pypika/terms.py]: ok, sounds like something we should eventually work out, but maybe not in this PR.

**Pull request merged**